Source/WebCore/platform/graphics/gstreamer/ImageGStreamerSkia.cpp

```diff
@@ -91,17 +87,19 @@ ImageGStreamer::ImageGStreamer(GRefPtr<GstSample>&& sample)
         break;
     }
 
+    m_size = { static_cast<float>(videoFrame->width()), static_cast<float>(videoFrame->height()) };
+
     auto toSkiaColorSpace = [](const PlatformVideoColorSpace&) {
         notImplemented();
         return SkColorSpace::MakeSRGB();
     };
-    auto imageInfo = SkImageInfo::Make(videoFrame->width(), videoFrame->height(), colorType, alphaType, toSkiaColorSpace(videoColorSpaceFromInfo(videoInfo)));
-    SkPixmap pixmap(imageInfo, videoFrame->planeData(0), videoFrame->planeStride(0));
-    auto image = SkImages::RasterFromPixmap(pixmap, [](const void*, void* context) {
-        std::unique_ptr<GstMappedFrame> videoFrame(static_cast<GstMappedFrame*>(context));
-    }, videoFrame.release());
+    auto imageInfo = SkImageInfo::Make(videoFrame->width(), videoFrame->height(), colorType, alphaType, toSkiaColorSpace(videoColorSpaceFromInfo(*videoInfo)));
 
-    m_image = BitmapImage::create(WTFMove(image));
+    // Copy the buffer data. Keeping the whole mapped GstVideoFrame alive would increase memory
+    // pressure and the file descriptor(s) associated with the buffer pool open. We only need the
+    // data here.
+    SkPixmap pixmap(imageInfo, videoFrame->planeData(0), videoFrame->planeStride(0));
+    m_image = SkImages::RasterFromPixmapCopy(pixmap);
```

Comment:
I still don't understand how copying the data fixes a memory leak, if there's a leak we are just moving the leaked data. Gst tool no longer reports a leak, but the copied data is still leaked, if the SkImage was not freed before, it won't now and it own the copied data. I don't understand why we keep a global reference of the image decoder, so I'll leave that to calvaris. If the goal of this patch is just to silence the gst leak detector, I would at least rename the bug title.